src/helpers/postMessageRelayHelpers.ts

```diff
@@ -321,3 +370,292 @@ export const handleAuthenticationPostMessage = ({
     }
   }
 };
+
+export type GraphQLSubscriptionLibrary =
+  | 'subscriptions-transport-ws'
+  | 'graphql-ws';
+
+// @see https://www.typescriptlang.org/docs/handbook/advanced-types.html#exhaustiveness-checking
+export function assertUnreachable(x: never): never {
+  throw new Error(`Didn't expect to get here ${x}`);
+}
+
+export class SubscriptionClient {
```

Comment:
In studio - when someone closes the subscription panel (cancels to subscription) we send a `EXPLORER_SUBSCRIPTION_TERMINATION` message. That message is in this repo too - we need to listen for that and unsubscribe / cancel the subscription when we get that. You can check out the [dev rools repo for how we do this there](https://github.com/apollographql/apollo-client-devtools/blob/main/src/application/components/Explorer/Explorer.tsx#L125)